pkg/api/entries.go

```diff
@@ -272,54 +272,37 @@ func getEntryURL(locationURL url.URL, uuid string) strfmt.URI {
 
 // GetLogEntryByUUIDHandler gets log entry and inclusion proof for specified UUID aka merkle leaf hash
 func GetLogEntryByUUIDHandler(params entries.GetLogEntryByUUIDParams) middleware.Responder {
-	ctx := params.HTTPRequest.Context()
-
 	uuid, err := sharding.GetUUIDFromIDString(params.EntryUUID)
 	if err != nil {
 		return handleRekorAPIError(params, http.StatusBadRequest, err, "")
 	}
-	var tid int64
 	tidString, err := sharding.GetTreeIDFromIDString(params.EntryUUID)
-	if err != nil {
-		// If EntryID is plain UUID, assume no sharding and use ActiveTreeID. The ActiveTreeID
-		// will == the tlog_id if a tlog_id is passed in at server startup.
-		if err.Error() == "cannot get treeID from plain UUID" {
-			tid = api.logRanges.ActiveTreeID()
-		} else {
-			return handleRekorAPIError(params, http.StatusBadRequest, err, "")
-		}
-	} else {
-		tid, err = strconv.ParseInt(tidString, 16, 64)
+
+	// If treeID is found in EntryID, route to correct tree
+	if err == nil {
+		tid, err := strconv.ParseInt(tidString, 16, 64)
 		if err != nil {
 			return handleRekorAPIError(params, http.StatusBadRequest, err, "")
 		}
-	}
-	hashValue, _ := hex.DecodeString(uuid)
-
-	tc := NewTrillianClientFromTreeID(params.HTTPRequest.Context(), tid)
-	log.RequestIDLogger(params.HTTPRequest).Debugf("Retrieving UUID %v from TreeID %v", uuid, tid)
-
-	resp := tc.getLeafAndProofByHash(hashValue)
-	switch resp.status {
-	case codes.OK:
-	case codes.NotFound:
-		return handleRekorAPIError(params, http.StatusNotFound, fmt.Errorf("grpc error: %w", resp.err), "")
-	default:
-		return handleRekorAPIError(params, http.StatusInternalServerError, fmt.Errorf("grpc error: %w", resp.err), trillianUnexpectedResult)
+		resp, _ := RetrieveUUID(params, uuid, tid)
```

Comment:
We probably shouldn't hide this error, might be useful to return it for the user if it's non-nil

Reply:
This makes sense now when returning `(models.LogEntry, error)` from `RetrieveUUID`